src/DacpacTool/PackageDeployer.cs

```diff
@@ -192,118 +190,37 @@ private void HandleDacServicesMessage(object sender, DacMessageEventArgs args)
             }
         }
 
-        public void SetProperty(string key, string value)
+        public void SetDeployProperty(string deployProperty)
+            => this.SetDeployProperties(new[] { deployProperty });
+
+        public void SetDeployProperties(string[] deployProperties)
         {
-            try
+            foreach (var deployProperty in deployProperties)
```

Comment:
Is it just me, or does it still feel like we're duplicating some logic here that's also in the `PropertyParser` class now?

Reply:
Where do you see the duplication? The `foreach` ?

Reply:
Mmmh we could actually have only one method for setting properties in a `DacDeployOptions` object. I just have a doubt regarding this verification and if it's applicable during the build:

```
                if (databaseProperty.Name == "SqlCommandVariableValues")
                {
                    throw new ArgumentException("SQLCMD variables should be set using the --sqlcmdvar command line argument and not as a property.");
                }
```

I'll have a look in a few hours

Reply:
I've mutualized the `SetDeployProperties` method